src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1451,7 +1451,7 @@ void CodeGen::genSSE2Intrinsic(GenTreeHWIntrinsic* node)
             {
                 assert(baseType == TYP_INT || baseType == TYP_UINT || baseType == TYP_LONG || baseType == TYP_ULONG);
                 op1Reg = op1->GetRegNum();
-                emit->emitIns_R_R(ins, emitActualTypeSize(baseType), op1Reg, targetReg);
+                emit->emitIns_R_R(ins, emitActualTypeSize(baseType), targetReg, op1Reg);
```

Comment:
Was this an existing bug?

Reply:
the instruction that we were generating here was `xmm2i` and for it we were passing `dst` as `reg2`. After this PR we pass it as `reg1` as we do for all other instructions (dst first, source second).